service/controller/resource/drainerinitializer/resource.go

```diff
@@ -249,22 +249,32 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 				continue
 			}
 
-			_, err = r.g8sClient.CoreV1alpha1().DrainerConfigs(cr.GetNamespace()).Get(privateDNS, metav1.GetOptions{})
+			dc, err := r.g8sClient.CoreV1alpha1().DrainerConfigs(cr.GetNamespace()).Get(privateDNS, metav1.GetOptions{})
 			if errors.IsNotFound(err) {
 				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find drainer config for ec2 instance %#q", *instance.InstanceId))
-
+				// create drainerConfig for the instance
 				err := r.createDrainerConfig(ctx, cl, cr, *instance.InstanceId, privateDNS)
 				if err != nil {
 					return microerror.Mask(err)
 				}
-
 			} else if err != nil {
 				return microerror.Mask(err)
 			} else {
-				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found drainer config for ec2 instance %#q", *instance.InstanceId))
+				// if the cluster id or instance id does not match, delete the bad CR and recreate it
+				if dc.Labels[key.TagCluster] != key.ClusterID(&cl) || dc.Annotations[annotation.InstanceID] != *instance.InstanceId {
```

Comment:
No show stopper, but I would like to put this logic in a separate function that has a descriptive name.

Reply:
done